compiler-core/src/analyse.rs

```diff
@@ -831,8 +831,16 @@ impl<'a, A> ModuleAnalyzer<'a, A> {
                      name,
                      arguments: args,
                      documentation,
+                     deprecation: constructor_deprecation,
+                     attributes_location,
                  }| {
                     self.check_name_case(name_location, &name, Named::CustomTypeVariant);
+                    if constructor_deprecation.is_deprecated() {
+                        self.track_feature_usage(
+                            FeatureKind::ConstructorWithDeprecatedAnnotation,
+                            attributes_location.expect("Check value must exist and be deprecated"),
```

Comment:
Let's point to the constructor rather than the attribute, and then remove this field